src/components/actions/triggersAndActions.ts

```diff
@@ -76,17 +77,19 @@ export async function processTrigger(payload: TriggerPayload) {
   const templateId = await DBConnect.getTemplateIdFromTrigger(payload.table, payload.record_id)
 
   // Get Actions from matching Template (and match templateActionCode if applicable)
-  const actions = await (
-    await DBConnect.getActionsByTemplateId(templateId, trigger)
-  ).filter((action) => {
-    if (!event_code) return true
-    else return action.event_code === event_code
-  })
+  const actions = (await DBConnect.getActionsByTemplateId(templateId, trigger))
+    .filter((action) => {
+      if (!event_code) return true
+      else return action.event_code === event_code
+    })
+    .map((action) => (action.code !== 'alias' ? action : swapOutAliasedActions(templateId, action)))
+
+  const resolvedActions = await Promise.all(actions)
```

Comment:
nice, simple and direct. (maybe a comment about Promise.all, that swapOutAliasedActions returns a promise that might needs to be resolved)

Reply:
This used to be a "for...of" loop, which loops synchronously. Swapping it for a .filter and .map makes them all async, hence the Promise.all() afterwards.

So the Promise.all is due to the async looping, not the swapOutAliasedActions function specifically.

Added a comment though: 6a54efe4